fortanix-sgx-tools/src/bin/ftxsgx-runner-cargo.rs

```diff
@@ -0,0 +1,123 @@
+            let heap_size = config.package.metadata.fortanix_sgx.heap_size.unwrap_or(HEAP_SIZE).to_string();
+            let ssaframesize = config.package.metadata.fortanix_sgx.ssaframesize.unwrap_or(SSAFRAMESIZE).to_string();
+            let stack_size = config.package.metadata.fortanix_sgx.stack_size.unwrap_or(STACK_SIZE).to_string();
+            let threads = config.package.metadata.fortanix_sgx.threads.unwrap_or(THREADS).to_string();
+            let debug = config.package.metadata.fortanix_sgx.debug.unwrap_or(DEBUG);
+
+            let args: Vec<String> = env::args().collect();
+
+            let mut ftxsgx_elf2sgxs_command = Command::new("ftxsgx-elf2sgxs");
+            ftxsgx_elf2sgxs_command.arg(&args[1])
+                .arg("--heap-size")
+                .arg(heap_size)
+                .arg("--ssaframesize")
+                .arg(ssaframesize)
+                .arg("--stack-size")
+                .arg(stack_size)
+                .arg("--threads")
+                .arg(threads);
+            if debug {
+                ftxsgx_elf2sgxs_command.arg("--debug");
+            }
+            let mut ftxsgx_elf2sgxs_status = ftxsgx_elf2sgxs_command.status()
+                .expect("Unable to execute ftxsgx-elf2sgxs");
+
+            if !ftxsgx_elf2sgxs_status.success() {
+                println!("{:?}", ftxsgx_elf2sgxs_command);
+                process::exit(match ftxsgx_elf2sgxs_status.code() {
+                    Some(code) => code,
+                    None => 1
+                });
+            }
+
+            let mut bin_with_ext = String::new();
+            bin_with_ext.push_str(&args[1]);
+            bin_with_ext.push_str(".sgxs");
+
+            let mut sgxs_append_command = Command::new("sgxs-append");
+            sgxs_append_command.arg("-i")
+                .arg(&bin_with_ext);
+
+            let sgxs_append_status = sgxs_append_command.status()
+                .expect("Unable to execute sgxs-append");
+
+            if !sgxs_append_status.success() {
+                println!("{:?}", sgxs_append_command);
+                process::exit(match sgxs_append_status.code() {
+                    Some(code) => code,
+                    None => 1
+                });
+            }
+
+            let mut ftxsgx_runner_command = Command::new("ftxsgx-runner");
+            ftxsgx_runner_command.arg(&bin_with_ext);
+
+            let ftxsgx_runner_status = ftxsgx_runner_command.status()
+                .expect("Unable to execute ftxsgx-runner");
+
+            if !ftxsgx_runner_status.success() {
+                println!("{:?}", ftxsgx_runner_command);
+                process::exit(match ftxsgx_runner_status.code() {
+                    Some(code) => code,
+                    None => 1
+                });
+            }
+
+            process::exit(0);
+        }
+        None => {
+            println!("{} is not defined in the environment.", key);
+            process::exit(1);
```

Comment:
Why?

Reply:
I was not sure because in this case the sub-commands don't run, so they can't error out, and so parent process shouldn't have non-zero exit status. But I believe the inability to run the process also classifies as a error.

Reply:
Not running anything is definitely an error, yes.